torchgeo/datasets/geo.py

```diff
@@ -605,8 +605,8 @@ def __init__(
             res: resolution of the dataset in units of CRS
             transforms: a function/transform that takes input sample and its target as
                 entry and returns a transformed version
-            label_name: name of the dataset property that has the label to be
-                rasterized into the mask
+            label_fn: function to extract the label to be rasterized into the mask from
+                the dataset Feature.
```

Comment:
Will think about this. I'm not opposed, but want to make sure lambdas aren't too complicated for remote sensing researchers with limited Python skills.

Reply:
Let me know if this should be changed. Perhaps alternative where we have function like `VectorDataset.get_class_index(feat: fiona.Feature)` where default implementation returns `feat["properties"][self.label_name]` is better, then it can be overloaded to derive the feature differently without needing to overload the entire `__getitem__`.

Reply:
Yeah, let's go with that alternative. Then, most datasets won't require this, we only need to think about it for particularly complicated datasets like EuroCrops.